File: code-samples/team-messaging/code-snippets/list-teams.php
<?php
list_teams("");

/*
* List teams under an account. Read 10 teams at a time.
*/
function list_teams($pageToken){
  global $platform;
  try{
    $queryParams = array(
      'recordCount' => 10,
      'pageToken' => $pageToken
    );

    $endpoint = "/team-messaging/v1/teams";
    $resp = $platform->get($endpoint, $queryParams);
    $jsonObj = $resp->json();
    // List teams API returns a list of teams in the ascending order based on team creation date and time.
    // I.e. from older team to newer team
    foreach ($jsonObj->records as $record) {
      // You can filter out any call queue you don't want to read analytics data!
      print_r ("The team " . $record->name . " was created on " . $record->creationTime . PHP_EOL);
    }
    // To read the next page, check and use the previous page token in the navigation object.
    if (property_exists($jsonObj->navigation, 'prevPageToken')){
        $pageToken = $jsonObj->navigation->prevPageToken;
        // Make sure not to exceed the API rate limit of 40 API calls per minute
        usleep(1200000);
        print_r ("Read newer teams ..." . PHP_EOL);
        list_teams($pageToken);
    }
  }catch (\RingCentral\SDK\Http\ApiException $e) {
    print_r ("Unable to read teams. " . $e->getMessage() . PHP_EOL);
  }
}
?>

File: code-samples/team-messaging/code-snippets/list-teams.java
package ListTeams;

import java.io.IOException;
import com.google.common.reflect.TypeToken;
import com.google.gson.Gson;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class ListTeams {
    static RestClient restClient;

    public static void main(String[] args) {
      var obj = new ListTeams();
      try {
        // Instantiate the SDK
        restClient = new RestClient("SANDBOX-APP-CLIENT-ID", "SANDBOX-APP-CLIENT-SECRET", "https://platform.devtest.ringcentral.com");

        // Authenticate a user using a personal JWT token
        restClient.authorize("SANDBOX-JWT");
        obj.list_teams();
      } catch (RestException e) {
        System.out.println(e.getMessage());
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
    /*
  	* List teams under an account. Read 10 teams at a time.
  	*/
  	public void list_teams(String pageToken) throws RestException, IOException {
  		try {
  			var queryParams = new ListGlipTeamsNewParameters()
  					.recordCount(10l)
  					.pageToken(pageToken);

  			var resp = restClient.teamMessaging().v1().teams().list(queryParams);
  			// List teams API returns a list of teams in the ascending order based on team creation date and time.
  			// I.e. from older team to newer team
  			for (var record : resp.records) {
  				System.out.println("Team \"" + record.name + " was created on " + record.creationTime);
  			}
  			// To read the next page, check and use the previous page token in the navigation object.
  			if (resp.navigation.prevPageToken != null) {
  				try {
  					Thread.sleep(1200);
  					System.out.println("Read newer teams ...");
  					list_teams(resp.navigation.prevPageToken);
  				} catch (InterruptedException e) {
  					e.printStackTrace();
  				}
  			}
  		}catch (RestException e){
  			System.out.println(e.getMessage());
  		}
  	}
}
